ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol.hs

```diff
@@ -447,7 +448,8 @@ mkShelleyGlobals epochInfo TPraosParams {..} = SL.Globals {
 -- | Check whether this node meets the leader threshold to issue a block.
 meetsLeaderThreshold
   :: forall c.
-     ConsensusConfig (TPraos c)
+     (VRF.VRFAlgorithm (VRF c))
```

Comment:
We use either `Crypto c` or `TPraosCrypto c` throughout `ouroboros-consensus-shelley`. I know you only need VRF here, but for consistency I would go for one of those other constraints.